apps/arweave/src/ar_chunk_storage.erl

```diff
@@ -1036,12 +1068,69 @@ get_packing_label(Packing, State) ->
 			{Label, State}
 	end.
 
+fix_broken_chunk_storage_record(ChunkFileStart, Start, PaddedEndOffset, State) ->
+	{Position, ChunkOffset} =
+		get_position_and_relative_chunk_offset_by_start_offset(ChunkFileStart, Start),
+	BucketStart = get_chunk_bucket_start(PaddedEndOffset),
+	case BucketStart + (ChunkOffset rem (?DATA_CHUNK_SIZE)) == PaddedEndOffset of
+		false ->
+			?LOG_WARNING([{event, fix_broken_chunk_storage_record_offset_mismatch},
+					{bucket_start, BucketStart},
+					{chunk_offset, ChunkOffset},
+					{padded_end_offset, PaddedEndOffset},
+					{computed_padded_end_offset,
+							BucketStart + (ChunkOffset rem (?DATA_CHUNK_SIZE))}]);
+		true ->
+			fix_broken_chunk_storage_record2(
+					ChunkFileStart, Position, PaddedEndOffset, State)
+	end.
+
+fix_broken_chunk_storage_record2(ChunkFileStart, Position, PaddedEndOffset, State) ->
+	StoreID = State#state.store_id,
+	case ar_sync_record:is_recorded(PaddedEndOffset, ar_chunk_storage, StoreID) of
+		false ->
+			fix_broken_chunk_storage_record3(ChunkFileStart, Position, State);
+		_ ->
+			?LOG_WARNING([
+				{event, requested_broken_chunk_storage_record_fix_for_valid_offset},
+				{chunk_file_start, ChunkFileStart},
+				{position, Position},
+				{padded_end_offset, PaddedEndOffset}])
+	end.
+
+fix_broken_chunk_storage_record3(ChunkFileStart, Position, State) ->
+	StoreID = State#state.store_id,
+	Filepath = filepath(ChunkFileStart, StoreID),
+	case file:open(Filepath, [read, write, raw]) of
+		{ok, F} ->
+			case file:pwrite(F, Position, << 0:(?OFFSET_BIT_SIZE) >>) of
```

Comment:
This will invalidate the chunk on disk, but leave the sync records intact, right? So a miner will have to run verify or something similar in order to "heal" the chunk on disk?

Reply:
This fixes the broken state when the chunk is found in the chunk file, but not in the ar_chunk_storage sync record so there is nothing left to heal. Hm, actually, it may be worth overwriting the entire chunk space with 0s here because the content may be potentially blocklisted.

Reply:
> This fixes the broken state when the chunk is found in the chunk file, but not in the ar_chunk_storage sync record

Ah, got it.